src/runtime_src/driver/xclng/tools/xbutil/xbutil.h

```diff
@@ -744,9 +746,33 @@ class device {
             ostr << "\n";
         }
         ostr << std::right << std::setw(80) << std::setfill('#') << std::left << "\n";
-        ostr << std::setfill(' ');
+        ostr << std::setfill(' ')<< "\n";
 #endif // AXI Firewall
+        xclDeviceUsage devstat = { 0 };
+        (void) xclGetUsageInfo(m_handle, &devstat);
+
+        m_mem_usage_stringize_dynamics(devstat, m_devinfo, usage_lines);
+        for(auto line:usage_lines){
+            ostr << line << "\n";
+        }
+        str_topology_print(ostr);
+        xclbinID_print(ostr);
+        return 0;
+    }
+
+
+    //STR topology
```

Comment:
@hcneema are we calling this "Stream Topology"? If so, let's rename this to printStreamTopology(). If not, let's rename to printStreamInfo() or something like that.